app/models/user.rb

```diff
@@ -466,9 +466,9 @@ def banished?
   end
 
   def subscribe_to_mailchimp_newsletter
-    return unless registered
-    return unless email.present? && email.include?("@")
-    return if saved_changes["unconfirmed_email"] && saved_changes["confirmation_sent_at"]
+    return unless registered && email.present?
+    return if SiteConfig.mailchimp_api_key.blank? && SiteConfig.mailchimp_newsletter_id.blank?
```

Comment:
**tiny optimization**: if we require both to be present, you can bail with the OR condition, `||`. The reason being that if `mailchimp_api_key` is empty we don't need to check the other one. This would potentially save us a roundtrip to the DB to load a configuration file. 

But if we only require one of the keys, discard my suggestion :D

Reply:
Hmm I guess we require both, but the guard is really just to see if Mailchimp is available... Theoretically a Forem admin never puts _just_ an API key, unless they forget to put the newsletter ID. 🤔

Reply:
Have you considered adding a tiny model that is not DB-backed? Examples for this include `Campaign` and `ForemInstance`.

You could then add then add a `subscribe` method there (and maybe a corresponding `unsubscribe` later) and pass in the user.

```ruby
class MailchimpNewsletter
  def self.subscribe(user)
     # all the logic here
  end
end
```

Then you could completely remove this method and the `after_save` would change to this:

```ruby
after_save -> { MailchimpNewsletter.subscribe(self) }
```

This hands-off the responsibility to decide whether or not a user actually should be subscribed to a dedicated object while also removing some code from `User`, which is too big anyway. 

If you don't like the plain Ruby model, a service could also work, e.g.

```ruby
after_save -> { MailchimpNewsletter::subscribe.call(self) }
```

Reply:
@citizen428 agreed that the user model is already too big.

The subscribing and unsubscribing currently happens in a service `Mailchimp::Bot` `app/services/mailchimp/bot.rb`. I hesitate to refactor that service in this PR since there's a lot of moving parts. Also, I'd rather keep this PR small because my main motivation was to prevent background jobs from being created on other Forems that don't have Mailchimp enabled.

Reply:
Makes sense @Zhao-Andy, maybe we can revisit this later, the `User` model in its current form has become a bit of a nemesis for me 😉